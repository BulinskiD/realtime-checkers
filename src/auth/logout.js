import React from 'react';
import Button from 'react-bootstrap/Button';
import {connect} from "react-redux";
import { userLogout } from "../store/actions/auth";

export const Logout = (props) => {
    return (
      <Button variant="danger" onClick={props.userLogout}>Logout</Button>
    );
}

export default connect(null, {userLogout})(Logout);
