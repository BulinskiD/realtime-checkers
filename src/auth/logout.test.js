import React from 'react';
import {shallow} from 'enzyme';
import {Logout} from './logout';

describe("Logout", () => {

    const userLogout = jest.fn();

    it('Should render button with event listener', () => {
       const component = shallow(<Logout userLogout={userLogout} />);
       component.find('Button').simulate('click');
       expect(userLogout).toHaveBeenCalledTimes(1);
       expect(component).toMatchSnapshot();
    });
});