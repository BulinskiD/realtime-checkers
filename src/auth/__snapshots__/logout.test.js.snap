// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Should render button with event listener 1`] = `
<Button
  active={false}
  disabled={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  type="button"
  variant="danger"
>
  Logout
</Button>
`;
